devices/flash-zynq7000/flashdrv.c

```diff
@@ -0,0 +1,461 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * Zynq-7000 nor flash driver
+ *
+ * Copyright 2020 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "qspi.h"
+#include "flashcfg.h"
+
+#include <lib/lib.h>
+#include <devices/devs.h>
+
+
+#define TIMEOUT_CMD_MS 0x05
+#define MAX_SIZE_CMD   0x20
+
+
+struct {
+	/* Buffers for command transactions */
+	u8 cmdRx[MAX_SIZE_CMD];
+	u8 cmdTx[MAX_SIZE_CMD];
+
+	/* Data caching in sector's buffer */
+	u8 *buff;
+	u32 buffPos;
+	u32 regID;
+	u32 sectID;
+
+	flash_info_t info;
+} fdrv_common;
+
+
+/* Auxiliary flash commands */
+
+static int flashdrv_cfiRead(flash_cfi_t *cfi)
+{
+	ssize_t res;
+	size_t cmdSz, dataSz;
+	/* RDID instruction is a generic for each flash memory */
+	const flash_cmd_t cmds = { .opCode = FLASH_CMD_RDID, .size = 1, .dummyCyc = 24 };
+
+	/* Cmd size is rounded to 4 bytes, it includes dummy cycles [b]. */
+	cmdSz = (cmds.size + cmds.dummyCyc / 8 + 0x3) & ~0x3;
+	/* Size of the data which is received during cmd transfer */
+	dataSz = cmds.dummyCyc / 8;
+
+	hal_memset(fdrv_common.cmdTx, 0, cmdSz);
+	fdrv_common.cmdTx[0] = cmds.opCode;
+
+	qspi_start();
+	if ((res = qspi_polledTransfer(fdrv_common.cmdTx, fdrv_common.cmdRx, cmdSz, TIMEOUT_CMD_MS)) < 0)
+		return res;
+
+	/* Copy data received after dummy bytes */
+	hal_memcpy((u8 *)cfi, (const void *)(fdrv_common.cmdRx + cmds.size), dataSz);
```

Comment:
Is this cast to `const void *` necessary?

Reply:
Nope. Removed.